nameable.rb

```diff
@@ -1,5 +1,5 @@
 class Nameable
   def correct_name
-    raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
+    raise NotImplementedError
   end
 end
```

Comment:
I can't highlight the relevant codes to buttress my point, but I will explain it here.

 - [ ] The essence of the Single Responsibility Rule is to ensure that functions/classes are responsible for a single job or thing. For example, in this application, the method for the user's input and input validation should be a different method from the method that creates the person or book. I would advise that you put this in a different method and call the method here. Kindly refer to this [documentation](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) 👍 💻 

In your `App` class, the following methods should have the code for taking user input different from the ones that create the object (instance of the class):
- `create_a_perspon`
- `create_student`
- `create_teacher`
- `create_a_rental`
- `create_a_book`

#### For example, in the `create_a_book` method, the set of codes in the screenshot below should be put in a separate method or module:

![image](https://user-images.githubusercontent.com/77106962/214131399-3232621d-770f-4e5e-a1a3-d63017379a1a.png)

#### While these should be put in a separate method or module

![image](https://user-images.githubusercontent.com/77106962/214131734-133fad40-2584-40c8-881b-8b2ce7d2bd12.png)


#### Hint: I would advise you to create helper modules for each scenario. 👍

Reply:
Hi @Gambit142 
Thank you for reviewing our project. Even though I studied the provided documentation many times and did not get the whole idea I  will be appreciated it if you refer us to some other tutorials or documentation.